lib/temporal/workflow/context.rb

```diff
@@ -280,11 +280,26 @@ def now
         state_manager.local_time
       end
 
-      def on_signal(&block)
+      # Define a signal handler to receive signals onto the workflow. When
+      # +name+ is defined, this creates a named signal handler which will be
+      # invoked whenever a signal named +name+ is received. A handler without
+      # a set name (defaults to nil) will be the default handler and will receive
+      # all signals that do not match a named signal handler.
+      #
+      # @param signal_name [String, Symbol, nil] an optional signal name; converted to a String
+      def on_signal(signal_name=nil, &block)
         target = History::EventTarget.workflow
 
-        dispatcher.register_handler(target, 'signaled') do |signal, input|
-          call_in_fiber(block, signal, input)
+        if signal_name
+          event_name = "signaled:#{signal_name}"
+          dispatcher.register_handler(target, event_name) do |_, input|
+            # do not pass signal name when triggering a named handler
+            call_in_fiber(block, input)
+          end
+        else
+          dispatcher.register_handler(target, Dispatcher::WILDCARD) do |signal, input|
```

Comment:
Something is missing here. This will basically call the handler on every single event dispatched on the `History::EventTarget.workflow`

Reply:
I was worried about that but unsure. You confirmed it was wrong. Just pushed a change more in line with your earlier guidance using `Dispatcher::TARGET_WILDCARD, 'signaled'` instead. Hopefully that won't get executed for all workflows whenever a signal arrives.